DEVELOPER.md

```diff
@@ -0,0 +1,103 @@
+# How to build the Docker image
+
+Each major version of the scanner gets its own image in a specific directory.
+
+Eg. to build sonar-scanner 4.x under the image name `scanner-cli`:
+
+```
+docker build --tag scanner-cli 4
+```
+
+# How to run the Docker image
+
+## On Linux
+
+### With local SonarQube
+
+With a SonarQube (SQ) running on default configuration (`http://localhost:9000`), the following will analyse the project in directory `/path/to/project`:
+
+```
+docker run -network=host --user="$(id -u):$(id -g)" -it -v "/path/to/project:/usr/src" sonarsource/sonar-scanner-cli
+```
+
+To analysis the project in the current directory:
+
+```
+docker run -network=host --user="$(id -u):$(id -g)" -it -v "$PWD:/usr/src" sonarsource/sonar-scanner-cli
+```
+
+If SQ is running on another port, you must specify it in your project's `sonar-project.properties` with `sonar.host.url=http://localhost:9010` and execute the same `docker run` command.
+
+### With SonarQube running in Docker
+
+First create a network and boot SonarQube:
+
+```
+docker network create "scanner-sq-network"
```

Comment:
why do we need to create a network ? the scanner can use `host.docker.internal`. Beside, i don't see uses cases where sq is running on the same docker host as the scanner.

Reply:
Or is this readme for sq developers ?

Reply:
`host.docker.internal` works only on mac (maybe on windows with Docker for Windows, but this has not been confirmed by @philippe-perrin-sonarsource )

> Or is this readme for sq developers ?

yes